pkg/scheduler/core/spreadconstraint/select_clusters.go

```diff
@@ -30,14 +30,20 @@ func SelectBestClusters(placement *policyv1alpha1.Placement, groupClustersInfo *
 func selectBestClustersBySpreadConstraints(spreadConstraints []policyv1alpha1.SpreadConstraint,
 	groupClustersInfo *GroupClustersInfo, needReplicas int32) ([]*clusterv1alpha1.Cluster, error) {
 	spreadConstraintMap := make(map[policyv1alpha1.SpreadFieldValue]policyv1alpha1.SpreadConstraint)
+	hasSpreadByLabel := false
 	for i := range spreadConstraints {
 		spreadConstraintMap[spreadConstraints[i].SpreadByField] = spreadConstraints[i]
+		if len(spreadConstraints[i].SpreadByLabel) > 0 {
+			hasSpreadByLabel = true
+		}
 	}
 
 	if _, exist := spreadConstraintMap[policyv1alpha1.SpreadByFieldRegion]; exist {
 		return selectBestClustersByRegion(spreadConstraintMap, groupClustersInfo)
 	} else if _, exist := spreadConstraintMap[policyv1alpha1.SpreadByFieldCluster]; exist {
 		return selectBestClustersByCluster(spreadConstraintMap[policyv1alpha1.SpreadByFieldCluster], groupClustersInfo, needReplicas)
```

Comment:
I have a question about the previous logic. 

Here bypass the logic according to `spreadByField`, what if we have two spread constraints? E.g: 
```yaml
    spreadConstraints:
      - spreadByField: region
        maxGroups: 2
        minGroups: 2
      - spreadByField: cluster
        maxGroups: 2
        minGroups: 2
```
This example has two constraints, one is `spreadByField: region` the other one is `spreadByField: cluster`.  For this case, we are going to select the cluster only by `region`?

Reply:
Oh, I get it.

https://github.com/karmada-io/karmada/blob/50cb376acd2b4838d14ed9f6b5d58f34f6765d18/pkg/scheduler/core/spreadconstraint/select_clusters_by_region.go#L21

Here checked the cluster group.

Reply:
Sorry, I didn't understand the purpose of your question. 'spreadByField' and 'spreadByLabel' cannot co-exist.